src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -102,6 +102,16 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
 
             var originalPackageSpec = matchingPackageSpecs.FirstOrDefault();
 
+            // Check if the project files are correct for CPM
+            if (originalPackageSpec.RestoreMetadata.CentralPackageVersionsEnabled)
+            {
+                var isValid = msBuild.AreCentralVersionRequirementsSatisfied(packageReferenceArgs);
+                if (!isValid)
+                {
+                    return 1;
```

Comment:
should we throw an exception instead of returning error exit code i.e. 1

Reply:
The `AreCentralVersionRequirementsSatisfied` method will log an error